### DISCUSSION TITLE
Fluentbit Flush configuration setting under Service

### DISCUSSION THREAD

**Discussion by rajagopr** [fluent/fluent-bit#5173] [Q&A] [answered]
I'm using the cloudwatch_logs output plugin. At what granularity would the following setting work?

    [SERVICE]
        Flush                     5
        Log_Level                 debug

If a given OUTPUT section writes data into N log streams in cloudwatch, at what granularity would the Flush setting apply?

1) Every 5 seconds for a given log group 
2) Every 5 seconds for a given log stream
3) Every 5 seconds for a given output section

Thanks

**Comment by agup006** [comment, marked as answer]
Hey @rajagopr , 3 fits the most here. Fluent Bit would chunk all incoming data and then flush that to all respective outputs at the flust interval (in this case every 5 seconds).
